common/src/main/java/com/wynntils/core/chat/tabs/ChatTabModel.java

```diff
@@ -74,6 +84,13 @@ public static void setFocusedTab(ChatTab focused) {
             chatTabData.putIfAbsent(focusedTab, new ChatComponent(McUtils.mc()));
             unreadMessages.put(focusedTab, false);
             McUtils.mc().gui.chat = chatTabData.get(focusedTab);
+
+            // If chat screen is open, and current message is empty, set our auto command
+            if (McUtils.mc().screen instanceof ChatScreen chatScreen
+                    && chatScreen.input.getValue().isEmpty()) {
+                String autoCommand = focusedTab.getAutoCommand() == null ? "" : focusedTab.getAutoCommand();
```

Comment:
Stop me if you think I'm too demanding here, but I think this is still not really ideal. Let's say you open the chat on a tab with no auto command. Then you tab your way to "Party" and get an autocommand of `/p`. Now you tab back to  "Local", but you will still keep the `/p`. And instead, if you were to continue switching tabs to "Guild", the `/p` would not be switched to a `/g`.

Reply:
Ok, now I realized what I want you to do. :-) Whenever you set an autocommand, you should also store it as a `lastAutocommand`, and when you get here, you should check if the input field is equal to the `lastAutocommand`. If it is, you should replace it with the new tabs autocommand. But if it is not equal to that (or empty), then you should leave it be.

Reply:
> Stop me if you think I'm too demanding here, but I think this is still not really ideal. Let's say you open the chat on a tab with no auto command. Then you tab your way to "Party" and get an autocommand of `/p`. Now you tab back to  "Local", but you will still keep the `/p`. And instead, if you were to continue switching tabs to "Guild", the `/p` would not be switched to a `/g`.

This didn't used to happen until i made this change, and I forgot to fix it again😅

Reply:
https://github.com/Wynntils/Artemis/pull/777/commits/8c50784ed2fd4d68dfa63aa750fd9ecd214e5352 even more elegant